### PR TITLE
Add Logging and Refactor Code

### PR DESCRIPTION
**Pull request by EangJS**
#54 
#46 

### Modify .gitignore
* Ignore ALL ```logging``` files
* Ignore ```htmlReport``` folder

### Add logging to Storage package
* Log messages are set to be display on the console and also saved into a ```logging.xml``` file
* Log all events that occur in StorageHandler

### Refactor individual storage class i.e. ```Load``` / ```Write UserData.java``` into ```StorageHandler.java```
* The read/write classes were their own individual classes. The classes can now be accessed through the StorageHandler class.

### Refactor UserCareerData.java from storage to userdata package
* It is more intuitive to have ```UserCareerData.java``` under userdata package rather than storage as it does not handle storage elements at all

### Add more JUnit test for storage package to increase testing coverage
* Check the deserializer and serializer of datetime
* Add test to handle case where ```userData.json``` file is missing

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -10,6 +10,7 @@ repositories {
 }
 
 dependencies {
+    implementation 'org.jetbrains:annotations:24.0.0'
```
Comment (ChubbsBunns) [inline build.gradle:13]: Should this dependency be extracted out? Did not see this being used,

File: src/main/java/seedu/duke/Duke.java [hunk 2]
```diff
@@ -1,46 +1,43 @@
 package seedu.duke;
 
 import seedu.duke.commands.CommandHandler;
-import seedu.duke.exceptions.DukeError;
 import seedu.duke.exercisegenerator.GenerateExercise;
-import seedu.duke.storage.LoadUserData;
-import seedu.duke.storage.UserCareerData;
+import seedu.duke.storage.StorageHandler;
```
Comment (ChubbsBunns) [inline src/main/java/seedu/duke/Duke.java:5]: I like the refactoring of the loading into storage handler

File: src/main/java/seedu/duke/Duke.java [hunk 3]
```diff
@@ -1,46 +1,43 @@
...
     private final ExerciseStateHandler exerciseHandler;
     private UserCareerData userCareerData;
+    private final StorageHandler storageHandler;
 
-    public Duke() {
-        exerciseHandler = new ExerciseStateHandler();
+    public Duke () {
         ui = new Ui();
         exerciseGenerator = new GenerateExercise();
-        try {
```
Comment (ChubbsBunns) [inline src/main/java/seedu/duke/Duke.java:24]: This usage of storage handler is good as it cleans the code

File: src/main/java/seedu/duke/commands/CommandHandler.java [hunk 4]
```diff
@@ -13,14 +14,15 @@ public class CommandHandler {
     private static final boolean INCOMPLETE_EXERCISE = false;
 
     /**
-     * @param rawUserCommands      This refers to the
-     * @param ui                   This allows us to output messages
-     * @param exerciseGenerator    This takes in filter parameters and outputs a curated exercise list
-     * @param userCareerData       This keeps track and allows logging of all user data
+     * @param rawUserCommands This refers to the
```
Comment (ChubbsBunns) [inline src/main/java/seedu/duke/commands/CommandHandler.java:17]: Is this an incomplete user command? This might be my mistake though. Will fix.

Comment (EangJS) [reply src/main/java/seedu/duke/commands/CommandHandler.java:17]: This one was like that before, nvr touch the javadoc there

**Review by ChubbsBunns — APPROVED** [review]
Good refactoring of the storage handling files
LGTM

**Pull request merged**